Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,213 @@
+﻿using Stratis.SmartContracts;
+using System;
+
+[Deploy]
+public class DAOContract : SmartContract
+{
+    public Address Owner
+    {
+        get => State.GetAddress(nameof(Owner));
+        private set => State.SetAddress(nameof(Owner), value);
+    }
+    /// <summary>
+    /// Min quorum for yes votes
+    /// </summary>
+    public uint MinQuorum
+    {
+        get => State.GetUInt32(nameof(MinQuorum));
+        private set => State.SetUInt32(nameof(MinQuorum), value);
+    }
+
+    public uint LastProposalId
+    {
+        get => State.GetUInt32(nameof(LastProposalId));
+        private set => State.SetUInt32(nameof(LastProposalId), value);
+    }
+
+    public bool IsWhitelisted(Address address) => State.GetBool($"Whitelisted:{address}");
+    private void SetIsWhitelisted(Address address, bool allowed) => State.SetBool($"Whitelisted:{address}", allowed);
+
+    public ulong GetVotingDeadline(uint proposalId) => State.GetUInt64($"Deadline:{proposalId}");
+    private void SetVotingDeadline(uint proposalId, ulong block) => State.SetUInt64($"Deadline:{proposalId}", block);
+
+    public uint GetYesVotes(uint proposalId) => State.GetUInt32($"YesVotes:{proposalId}");
+    private void SetYesVotes(uint proposalId, uint value) => State.SetUInt32($"YesVotes:{proposalId}", value);
+
+    public uint GetNoVotes(uint proposalId) => State.GetUInt32($"NoVotes:{proposalId}");
+    private void SetNoVotes(uint proposalId, uint value) => State.SetUInt32($"NoVotes:{proposalId}", value);
+
+    public uint GetVote(uint proposalId, Address address) => State.GetUInt32($"Vote:{proposalId}:{address}");
+
+    public Proposal GetProposal(uint index) => State.GetStruct<Proposal>($"Proposals:{index}");
+
+    private void SetProposal(uint index, Proposal proposal) => State.SetStruct($"Proposals:{index}", proposal);
+
+    public DAOContract(ISmartContractState state, uint minQuorum)
+        : base(state)
+    {
+        this.MinQuorum = minQuorum;
```

Comment:
MinQuorum will need a method that the owner can use to update since it looks likes its not percentage based. If it is percentage based, will need to update the assert methods during proposal execution. 

Another thought would be to use a percentage here, and keep track of the number of active whitelisted addresses to calculate the total required amount of yes votes for a proposal to succeed. 

As Masternodes leave/join, will need to make sure that this value is properly set at the right times.

Reply:
This is just first version and we wanted to keep things simpler.

Reply:
You still will need a way for this to be updated if you haven't added that. For example if there are 100 whitelisted addresses when the contract is created and `MinQuorum = 51` then masternodes leave or are blacklisted, the quorum needs to be changed.

Reply:
You can keep track of the total whitelisted addresses and do something like `MinQuorum = whitelistedCount / 2 + 1` to adjust this dynamically.

Reply:
> I'd recommend not marking discussions as resolved until code is adjusted accordingly or an agreement on the suggestion has been made in the future 👍 left some more comments on some of the resolved comments.

Ok I will leave marking it as resolved to you since now.

Reply:
Firstly, already yes votes should be higher than no votes. 
Secondly, if there is 2 person yet whitelisted, that doesn't mean that 2 of them can accept proposal and execute it. MinQuorum is minimum amount of Yes votes needed for executing proposal that doesn't strongly dependant on number of voters. The owner can change the value later always if needed. 

I will leave it as it is for now.